src/components/DataViewer/DataViewer.tsx

```diff
@@ -86,15 +115,10 @@ const DataViewer: React.FC<DataViewerProps> = (props) => {
     const closeModal = (): void => {
         props.onRequestClose();
     };
-    if (!props.isOpen) {
-        return <></>;
-    }
-
-    Modal.setAppElement("#modal-parent");
 
     return (
         <StyledModal
-            isOpen={true}
+            isOpen={props.isOpen}
```

Comment:
Is there a reason for changing this? 

I think I prefer the previous implementation of 
```if (!props.isOpen) {
    return <></>;
}```
because it means the modal will not be rendered if it doesn't have to, whereas now it is still rendered in the background even if it's not closed.

Reply:
should the modal be rendered and 'not shown' for screen-readers and accessibility reasons? having rerenders of the page may be worse than just rendering everything once an dynamically adjusting visibility rules.

(for the record, I don't know what is 'better'! we should Google and decide as a team what's 'better')

Reply:
We looked into the html in the element - the only difference is if we use props.isOpen then there will be an empty react modal div generated when it is closed. It doesn't seem to bring much difference and also the code is tidier when we directly put props.isOpen inside the modal

Reply:
There are likely a lot of ARIA labels which help the screen-readers etc out when it is closed - there's a reason they provide an `isOpen` attribute instead of just saying 'only have this on your page if needed'!

Reply:
Also apparently there may be a 'better closing animation' when you close the page - may be worth checking!